processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -383,12 +411,18 @@ else if ( !method.isUpdateMethod() ) {
                 returnTypeBuilder,
                 beforeMappingMethods,
                 afterMappingMethods,
+                beforeMappingReferencesWithFinalizedReturnType,
+                afterMappingReferencesWithFinalizedReturnType,
                 finalizeMethod,
                 mappingReferences,
                 subclasses
             );
         }
 
+        private void removeMappingReferencesWithoutSourceParameters(List<LifecycleCallbackMethodReference> references) {
+            references.removeIf( r -> r.getSourceParameters().isEmpty() && r.getReturnType().isVoid() );
```

Comment:
Let's add a comment why we are doing this. The reason why this is done is because lifecycle methods without any parameters are going to be already invoked. Mostly for future us, when we scratch our head why this is being done 😄

Reply:
Thank you for explaining why I did this. I already already forgot the reason 2 weeks after implementing it... 😆  I added a comment.